### PR TITLE
[RLlib] Fix 2 broken CI test cases: `test_learner_group` and `cartpole_dqn_envrunner`.

### PR DESCRIPTION
**Pull request by sven1977**
Fix 2 broken CI test cases: `test_learner_group` and `cartpole_dqn_envrunner`.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/env/single_agent_env_runner.py [hunk 1]
```diff
@@ -274,7 +274,7 @@ def _sample_timesteps(
                 if explore:
                     env_steps_lifetime = self.metrics.peek(
                         NUM_ENV_STEPS_SAMPLED_LIFETIME
```
Comment (simonsays1980) [inline rllib/env/single_agent_env_runner.py:276]: `NUM_ENV_STEPS_SAMPLED_LIFETIME` is then update during synching the metrics in between sampling?

**Review by simonsays1980 — APPROVED** [review]
LGTM.

File: rllib/env/single_agent_env_runner.py (hunk 1 above)
Comment (sven1977) [reply rllib/env/single_agent_env_runner.py:276]: Correct, it's always updated anyways (this was a bug) after each single env step. Inside `self._increase_sampled_metrics()`.

Comment (sven1977) [reply rllib/env/single_agent_env_runner.py:276]: And it's also hard-overwritten by the Algo itself after each iteration in order to sum up all the steps from all the other EnvRunners (making sure each EnvRunner always has the total steps sampled, not just its own lifetime count).

**Pull request merged**